### PR TITLE
Update naxis after map.resample

### PR DESCRIPTION
**Pull request by lzivadinovic**
Description

This PR updates NAXISn keywords based on resampled data shape.

### CODE REVIEW

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -1202,6 +1202,8 @@ def resample(self, dimensions: u.pixel, method='linear'):
         lon, lat = self._get_lon_lat(self.center.frame)
         new_meta['crval1'] = lon.value
         new_meta['crval2'] = lat.value
+        new_meta['naxis1'] = new_data.shape[1]
+        new_meta['naxis2'] = new_data.shape[0]
```
Comment (nabobalis) [inline sunpy/map/mapbase.py:1206]: There might be a rotate test that can be updated to check that this is set.

Comment (lzivadinovic) [reply sunpy/map/mapbase.py:1206]: ahhh yes, i was loking for it, but didnt see the one that tests metadata.

https://github.com/sunpy/sunpy/blob/6e40e8edce7fe0e75753620417fb51452db1d5e0/sunpy/map/tests/test_mapbase.py#L579

I will add it.

**Review by dstansby — APPROVED** [review]
Looks 👍 if tests pass. Does this fix a reported issue?

File: changelog/4445.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixes issue where NAXISn is not updated after invoking map.resample()
```
Comment (Cadair) [inline changelog/4445.bugfix.rst:1]:
```suggestion
Fixes issue where NAXISn is not updated after invoking :method:`.GenericMap.resample`
```

**Review by Cadair — APPROVED** [review]

**Comment by dstansby** [comment]
Test failures are un-related

**Pull request merged**